File: src/main/java/me/aikoo/Karl/Interpreter/parser/ast/values/StringValue.java
package me.aikoo.Karl.Interpreter.parser.ast.values;

import me.aikoo.Karl.Interpreter.parser.TokenType;

public class StringValue extends Value {
    private final String value;

    public StringValue(String value) {
        this.value = value;
    }

    @Override
    public String toString() {
        return value;
    }

    @Override
    public int toInt() {
        return Integer.parseInt(value);
    }

    @Override
    public float toFloat() {
        return Float.parseFloat(value);
    }

    @Override
    public TokenType getType() {
        return TokenType.STR_VALUE;
    }
}

File: src/main/kotlin/studio/karllang/karl/parser/ast/values/StringValue.kt
package studio.karllang.karl.parser.ast.values

import studio.karllang.karl.lexer.TokenType

class StringValue(private val value: String) : Value() {
    override val type = TokenType.STR_VALUE

    override fun toString(): String {
        return value
    }

    override fun toInt(): Int {
        return value.toInt()
    }

    override fun toFloat(): Float {
        return value.toFloat()
    }
}